packages/graphql-typescript-definitions/src/index.ts

```diff
@@ -240,19 +338,119 @@ export class Builder extends EventEmitter {
     }
   }
 
-  private async updateDocumentForFile(file: string) {
-    const contents = await readFile(file, 'utf8');
+  private async updateDocumentForFile(filePath: string) {
+    const project = this.config.getConfigForFile(filePath);
+
+    if (!project) {
+      throw new Error(`No project found for file: ${filePath}`);
+    }
+
+    const projectName = project.projectName || '';
+
+    let documents = this.documentMapByProject.get(projectName);
+
+    if (!documents) {
+      documents = new Map<string, DocumentNode>();
+      this.documentMapByProject.set(projectName, documents);
+    }
+
+    const contents = await readFile(filePath, 'utf8');
+
     if (contents.trim().length === 0) {
-      return;
+      return undefined;
     }
 
-    const document = parse(new Source(contents, file));
-    this.documentCache.set(file, document);
+    const document = parse(new Source(contents, filePath));
+    documents.set(filePath, document);
+
+    return document;
   }
 
-  private removeDocumentForFile(file: string) {
-    this.documentCache.delete(file);
+  private getProjects() {
+    const projects = this.config.getProjects();
+
+    if (projects) {
+      return Object.values(projects);
+    }
+
+    const project = this.config.getProjectConfig();
+
+    if (project) {
+      return [project];
+    }
+
+    this.emit('error', new Error(`No projects configured in .graphqlconfig`));
```

Comment:
Personally, I'd rather validate this in the constructor and just fail as early as possible